CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1088,7 +1088,7 @@ public static function processRecurringContribution(&$form, &$params, $contactID
     }
     $recurParams['invoice_id'] = $params['invoiceID'] ?? NULL;
     $recurParams['contribution_status_id'] = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending');
-    $recurParams['payment_processor_id'] = $params['payment_processor_id'] ?? NULL;
+    $recurParams['payment_processor_id'] = $recurParams['processor_id'] = $params['payment_processor_id'] ?? NULL;
```

Comment:
I'm not 100% sure this is correct, 

The processor_id is meant to be the unique token or similar that is used when charging the payment. The payment_processor_id = the id of the PaymentProcessor that is used from civicrm_payment_processor table

Reply:
Makes sense. I figured the processor_id == payment_processor_id, which is not true.